### PR TITLE
Another refactor to simplify cache and keep consistency

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

**Related Issue:**  N/A

### Description of changes
- Continuation of 
  - #114 and 
  - #115 
- Simplify cache and manage data fields consistently using `cacheFieldMgmt{}`

### CODE REVIEW

**Review by Anxhela21 — APPROVED** [review]
/lgtm
/hold

File: pkg/rbac/sharedData.go [hunk 1]
```diff
@@ -27,15 +27,14 @@ type SharedData struct {
 	namespaces       []string
 	propTypes        map[string]string
 
-	// These are internal objects to track the state of the cache.
+	// Added fields to manage the state of the cached data.
 	csrCache    cacheFieldMgmt
```
[the remark this answers is not in the document]
Comment (jlpadilla) [reply pkg/rbac/sharedData.go]: renamed, thanks for the suggestion!

**Review by Anxhela21 — APPROVED** [review]
/lgtm

**Pull request merged**